middleware/http/ratelimit/ratelimit_middleware.go

```diff
@@ -54,7 +54,26 @@ func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(next http.Ha
 	limiter := tollbooth.NewLimiter(meta.MaxRequestsPerSecond, nil)
 
 	return func(next http.Handler) http.Handler {
-		return tollbooth.LimitHandler(limiter, next)
+		// Adapted from toolbooth.LimitHandler
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// We need to set a value for X-Forwarded-For or the rate limiter won't work
+			r.Header.Set("X-Forwarded-For", "0.0.0.0")
```

Comment:
Can this override a user defined X-Forwarded-For?

Reply:
In theory, yes. In practice, the requests that hit the Dapr sidecar only come from client apps and there shouldn't be a X-Forwarded-For - if I'm not mistaken

Reply:
Just to be safe, I changed the code to try and look up the IP (like the tollbooth library does), and only set "X-Forwarded-For" if strictly necessary.

Reply:
Thanks, this change makes sense.